nengo_loihi/emulator/interface.py

```diff
@@ -353,11 +353,9 @@ def __init__(self, block_info):
             self.target_u[sl] = compartment.noiseAtDendOrVm
 
         if self.dtype == np.int32:
-            if np.any(self.exp < 7):
-                warnings.warn("Noise amplitude falls below lower limit")
-                self.exp[self.exp < 7] = 7
-
-            self.mult = np.where(self.enabled, 2**(self.exp - 7), 0)
+            # TODO: if we could do this mult with shifts, it'd be faster, but
+            # numpy has no function taking a vector of positive/negative shifts
+            self.mult = np.where(self.enabled, 2.**(self.exp - 7), 0)
```

Comment:
Is this being tested anywhere for exp < 7?

Reply:
Why was the `self.exp[self.exp < 7] = 7` removed? I added it because it was happening all the time and crashes when it does.

Reply:
Well it shouldn't crash. That would be a bug with the emulator. Can you reproduce the crash?

The main reason I removed it is that we were getting different behaviour between the emulator and chip on a test. I believe that the chip will allow a noise exponent down to 0. It could be that I'm wrong, there, though. Either way, any clipping or warning about values should be done in `discretize.py` and `validate.py`, so that it happens for both emulator and chip. If the emulator specifically is crashing, then we need to fix it for these values.

Reply:
My guess is that the bug was because we had `2**(self.exp - 7)` instead of `2.**(self.exp - 7)`, so it wasn't allowing for floating point multiplication as needed. That's been changed here, so the bug should be resolved. We should make sure it's well tested though.

Reply:
In general, we don't have good tests for the noise. The one test that we have doesn't assert anything. I'll see if I can fix it up, since that's something we've wanted to do anyway, and it shouldn't be too difficult.

Reply:
Took a bit more time than I expected, because it revealed a bug that I ended up fixing. It fell loosely into my extra hour :)